src/mahoji/lib/abstracted_commands/minionKill.ts

```diff
@@ -369,18 +369,30 @@ export async function minionKillCommand(
 	}
 
 	function applyBlackMaskBoost() {
+		const hasInfernalSlayerHelmI = monster?.canBePked
+			? wildyGear.hasEquipped('Infernal slayer helmet(i)')
+			: user.hasEquippedOrInBank('Infernal slayer helmet(i)');
+		const hasInfernalSlayerHelm = monster?.canBePked
+			? wildyGear.hasEquipped('Infernal slayer helmet')
+			: user.hasEquippedOrInBank('Infernal slayer helmet');
 		const hasBlackMask = monster?.canBePked
 			? wildyGear.hasEquipped('Black mask')
 			: user.hasEquippedOrInBank('Black mask');
-		const hasBlackMaskI = isOnTask
+		const hasBlackMaskI = monster?.canBePked
 			? wildyGear.hasEquipped('Black mask (i)')
 			: user.hasEquippedOrInBank('Black mask (i)');
 
 		if (attackStyles.includes(SkillsEnum.Ranged) || attackStyles.includes(SkillsEnum.Magic)) {
```

Comment:
This doesn't take into account Infernal slayer helmet(i) while training melee, also L329 to L346 should be removed

Reply:
Fixed I think?

Reply:
You still want the checks for range and mage though, it should be something like this https://github.com/oldschoolgg/oldschoolbot/issues/5445#issuecomment-1773842000

Reply:
I miss read your code, my bad. I believe this will work!